src/main/java/uk/ac/ebi/tsc/portal/api/team/service/TeamService.java

```diff
@@ -907,4 +911,16 @@ public TeamResource populateTeamContactEmails(Team team, TeamResource teamResour
 		}
 		return teamResource;
 	}
+
+	public TeamResource populateTeamMemberEmails(Team team, TeamResource teamResource, String username) {
+		if (username.equals(team.getAccount().getUsername()) || teamResource.getManagerUserNames().contains(username)) {
+			teamResource.setMemberAccountEmails(team.getAccountsBelongingToTeam().stream().map(a -> a.getEmail()).collect(Collectors.toList()));
+		} else {
+			boolean isMember = team.getAccountsBelongingToTeam().stream().anyMatch(a -> a.getUsername().equals(username));
+			if (isMember) {
+				teamResource.getMemberAccountEmails().add(accountService.findByUsername(username).email);
```

Comment:
Why do we need to look up for the email of the member account by an extra call to DB, while we already know it?